addon/instance-initializers/prefetch.js

```diff
@@ -14,6 +14,9 @@ export function initialize(instance) {
     // If there is no pivot, we should try to prefetch all handlers.
     let hasSeenPivot = pivotHandler == null ? true : false;
 
+    // For asynchronously loaded handlers, we chain them to ensure
+    // resolution order.
+    transition._handlerPromiseChain = Ember.RSVP.resolve();
```

Comment:
Can the handler promise chain be stored in a local variable? Let's not muck with the transition object unless it's unavoidable.